### PR TITLE
Ninjonas userspace 

### PR DESCRIPTION
**Pull request by ninjonas**
## Description
- Kyria: moved OLED & Encoder code to separate files
- ninjonas.c: Created logic to cycle through hue wheel when starting keyboard
- ninjonas/oled.c: Minor OLED refactors

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kyria/keymaps/ninjonas/encoder.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "ninjonas.h"
+
+#ifdef ENCODER_ENABLE
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        switch (biton32(layer_state)) {
```
Comment (drashna) [inline keyboards/kyria/keymaps/ninjonas/encoder.c]:
```suggestion
        switch (get_highest_layer(layer_state)) {
```

File: keyboards/kyria/keymaps/ninjonas/encoder.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            default:
+                if (clockwise) {
+                    tap_code(KC_BRIU);
+                } else {
+                    tap_code(KC_BRID);
+                }
+                break;
+        }
+    } else if (index == 1) {
+        switch (biton32(layer_state)) {
```
Comment (drashna) [inline keyboards/kyria/keymaps/ninjonas/encoder.c]:
```suggestion
        switch (get_highest_layer(layer_state)) {
```

File: keyboards/kyria/keymaps/ninjonas/encoder.c (hunk 1 above)
Comment (ninjonas) [reply keyboards/kyria/keymaps/ninjonas/encoder.c]: thanks!

File: keyboards/kyria/keymaps/ninjonas/encoder.c (hunk 2 above)
Comment (ninjonas) [reply keyboards/kyria/keymaps/ninjonas/encoder.c]: thanks again!

**Comment by fauxpark** [comment]
Thanks!

**Pull request merged**